core/libraries/rest_api/calculations/CalculatedModelFieldsFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace EventEspresso\core\libraries\rest_api\calculations;
+
+use EventEspresso\core\domain\services\factories\FactoryInterface;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use EventEspresso\core\exceptions\UnexpectedEntityException;
+use InvalidArgumentException;
+use EventEspresso\core\services\loaders\Loader;
+
+/**
+ * Class CalculatedModelFieldsFactory
+ *
+ * Loads classes for calculating fields for the REST API
+ *
+ * @package     Event Espresso
+ * @author         Mike Nelson
+ * @since         $VID:$
+ *
+ */
+class CalculatedModelFieldsFactory
+{
+    private $loader;
+
+    /**
+     * CalculatedModelFieldsFactory constructor.
+     * @param Loader $loader
+     */
+    public function __construct(Loader $loader)
```

Comment:
This should still typehint for `LoaderInterface`.  `EE_Dependency_Map` can still register the actual loader for injection but you don't want to typehint the concrete here.

Reply:
ahhh I see